README.md

```diff
@@ -66,74 +64,84 @@ yarn test:e2e
 
 ## Usage
 
-Retrieving and generating the cache file have their own respective endpoint
-
-### Fetch a cache file
+### Votes CSV report
 
-Send a POST request with a proposal ID
+Generate and serve votes CSV report for closed proposals.
 
-```bash
-curl -X POST localhost:3000/votes/[PROPOSAL-ID]
-```
+> NOTE: CSV files are generated only once, then cached, making this service a cache middleware between snapshot-hub and UI
 
-When cached, this request will respond with a stream to a CSV file.
+#### Fetch a cache file
 
-On all other cases, it will respond with a [JSON-RPC 2.0](https://www.jsonrpc.org/specification) error response:
+Send a `POST` request with a proposal ID
 
 ```bash
-{
-  "jsonrpc":"2.0",
-  "error":{
-    "code": CODE,
-    "message": MESSAGE
-  },
-  "id": PROPOSAL-ID
-}
+curl -X POST localhost:3000/api/votes/[PROPOSAL-ID]
 ```
 
-| Description                         | `CODE` | `MESSAGE`           |
-| ----------------------------------- | ------ | ------------------- |
-| When the proposal does not exist    | -40001 | PROPOSAL_NOT_FOUND  |
-| When the proposal is not closed     | -40004 | PROPOSAL_NOT_CLOSED |
-| When the file is pending generation | -40010 | PENDING_GENERATION  |
-| Other/Unknown/Server Error          | -32603 | INTERNAL_ERROR      |
-
-Furthermore, when votes report can be cached, but does not exist yet, a cache generation task will be queued. This enable cache to be generated on-demand.
+When cached, this request will respond with a stream to a CSV file.
+When votes report can be cached, but does not exist yet, a cache generation task will be queued. This enable cache to be generated on-demand.
 
-### Generate a cache file
+#### Generate a cache file
 
-Send a POST request with a body following the [Webhook event object](https://docs.snapshot.org/tools/webhooks).
+Send a `POST` request with a body following the [Webhook event object](https://docs.snapshot.org/tools/webhooks).
 
 ```bash
-curl -X POST localhost:3000/votes/generate \
+curl -X POST localhost:3000/webhook \
```

Comment:
Should be /api/webhook no?

Reply:
I would like to keep /api only for output data, and /webhook outside it. wdyt ?

Reply:
makes sense as it is not really api